src/app/components/buscadorTurnosPrestaciones/turnos-prestaciones.component.ts

```diff
@@ -280,8 +285,70 @@ export class TurnosPrestacionesComponent implements OnInit {
         }
     }
 
+    exportPrestaciones() {
+        let prestacionesCheck = [];
+        this.prestacionesExport = [];
+        this.busqueda$.pipe(
+            map((prestaciones) => {
+                prestacionesCheck = prestaciones.filter(prestacion => prestacion.check === true);
+            })
+        ).subscribe();
+        if (prestacionesCheck.length) {
+            prestacionesCheck.forEach(elem => {
+                this.prestacionesExport.push(elem.idPrestacion);
+            });
+            this.exportHudsService.peticionHuds({ arrayPrestaciones: this.prestacionesExport }).subscribe(res => {
+                if (res) {
+                    this.plex.toast('success', 'Su pedido esta siendo procesado, diríjase a descargas pendientes para obtener su reporte', 'Información', 2000);
+                    this.getPendientes();
+                }
+            });
+        }
+
+    }
+
+    selectPrestacion(prestacionCeck) {
+        if (prestacionCeck) {
+            this.enableExport = true;
+        } else {
+            this.prestacionesAll = false;
+            let found;
+            this.busqueda$.pipe(
+                map((prestaciones) => {
+                    found = prestaciones.find(prestacion => prestacion.check === true);
+                })
+            ).subscribe();
+            this.enableExport = found ? true : false;
+        }
+    }
```

Comment:
Siempre pensá que los `.subscribe` no se usan en el 99% de los casos. 
Va un tip orientador:

```suggestion 
    this.enableExport$ = this.busqueda$.pipe(
        map(prestaciones => prestaciones.some(prestacion => prestacion.check === true) )
    );
```

Y en el HTML:

```
<plex-button label="Exportar" size="sm" type="success" class="mr-2" (click)="exportPrestaciones()"
                     [disabled]="! (enableExport$ | async) ">
 </plex-button>
```

Sugerencia: invertir la condición para no introducir la negación en el HTML que es más dificil de revisar e interpretar.

Reply:
Y si fuera el caso que tenes que usar subscribe, debería meter el código dentro de subscribe, RXJS no garantiza la sincronicidad de la ejecución. 
La linea 321 podría llegar a ejecutarse antes de lo que esta dentro del **map()**